### ISSUE TITLE
License concern forking this repo

### ISSUE THREAD

**Issue by pepsifan** [bbernhard/signal-cli-rest-api#154]
When looking for a simple Signal client to send notifications I stumbled over this project. It worked out of the box with the docker compose file (thanks for this project). As already mentioned by others there's some delay while calling the API (between 4-10 seconds). I guess it's related to signal-cli. One solution suggested is to use the dbus mode. So my naive approach would be:
- add `dbus` package to Dockerfile
- start the dbus daemon
- apply the variable `DBUS_SESSION_BUS_ADDRESS`
- start signal-cli in daemon mode
- add a new API method which sends messages over dbus

As mentioned in https://github.com/bbernhard/signal-cli-rest-api/issues/30#issuecomment-704503328 it is not planned to add dbus support in the near future. But for my needs it's sufficient to have only a subset of commands with dbus.

But now to the main topic why I started this issue. Before forking this project and make some changes, I verified the license model. It states that it use the MIT License, but in this git repository there are some libs which are licensed under the GPLv3 (zkgroup and libsignal-client). These libs are needed for signal-cli and the golang programm in this project is onlly making some console calls to signal-cli. So I guess there is no issue with GPL. But is it a issue to have lib files with GPL license in a MIT licensed repo (even without notice)? Would be nice if someone can clarify.

**Comment by bbernhard** [comment]
I am not an expert, but my understanding of this has always been that it is fine as long one is not statically linking against a GPL licensed software. As `zkgroup` and `libsignal-client` are both dynamically linked (those are just shared objects), I _think_ it should be fine. 

Regarding DBUS support: I've started working on a DBUS implementation a long time ago in the `dbus` git branch. It's probably not useful anymore though, as the `master` branch has changed too much ;).

In case you want to merge your changes back, I would definitely appreciate a PR for that. Although there's the native mode which improves performance quite a bit, there are still some use cases for which it is too slow. I think we could either create a second Dockerfile for that or use env variables to somehow switch between the different modes when the container starts. But I also totally understand if you want keep your changes on a separate fork - that's also perfectly fine for me :)

**Comment by pepsifan** [comment]
Thanks for the quick answer. Then it's fine to stay at the MIT License. Otherwise I would have suggested to download the libs during the Docker build. There's a nice repo providing the compiled versions: https://github.com/exquo/signal-libs-build

Concerning the performance issue, I already tried the native mode but it still takes several seconds. I also tried the dbus mode of `signal-cli` and it's really a huge difference. That's why I want to modify this project for my purposes.

But after some brainstorming, I think it would be better (and maybe easier) to change the code of `signal-cli` (add some rest endpoints) instead following the idea how to wrap the dbus mechanism :)
